File: 1065.java
import java.util.Scanner;

public class Main{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int inputNumber = sc.nextInt();
        if(inputNumber < 1 || inputNumber > 1000)   System.exit(0);
        else    System.out.println(checkNum(inputNumber));
    }
    public static int checkNum(int givenNumber){
        int counter = 0;
        for(int currentNumber = 1; currentNumber < givenNumber+1; currentNumber++){
            int j = currentNumber/1000;
            int k = (currentNumber-j*1000)/100;
            int m = (currentNumber-j*1000-k*100)/10;
            int n = currentNumber-j*1000-k*100-m*10;
            
            if(j == 0 && k == 0)  counter++;
            else if(j == 0 && k-m == m-n)  counter++;         
        }
    	return counter;
    }
}

File: 1065.py
def checkNumber(givenNumber):
    counter = 0
    for i in range(givenNumber):
        each = i + 1                                            #i starts from 0 
        j = int(each/1000)
        k = int((each-j*1000)/100)
        m = int((each-j*1000-k*100)/10)a
        n = int(each-j*1000-k*100-m*10)
        if j == 0 and k == 0:   counter = counter + 1
        elif j == 0 and k-m == m-n:  counter = counter + 1
    return counter
    
inputNumber = int(input())

if inputNumber < 1 or inputNumber > 1000:   sys.exit()
else :  print(checkNumber(inputNumber))
